packages/firestore/src/core/sync_engine.ts

```diff
@@ -1040,18 +1077,20 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
     }
 
     for (const targetId of removed) {
-      const queryView = this.queryViewsByTarget[targetId];
-      // Check that the query is still active since the query might have been
+      // Check that the target is still active since the target might have been
       // removed if it has been rejected by the backend.
-      if (queryView) {
-        await this.localStore
-          .releaseQuery(queryView.query, /*keepPersistedQueryData=*/ false)
-          .then(() => {
-            this.remoteStore.unlisten(targetId);
-            this.removeAndCleanupQuery(queryView);
-          })
-          .catch(ignoreIfPrimaryLeaseLoss);
+      if (!this.queriesByTarget[targetId]) {
```

Comment:
Above, you check for targetId existence and emptiness. It would be nice if we could guarantee that we only need to check for targetId existence.

Reply:
I don't feel strongly. If what you have works and it's consistent, then we can probably keep it as is. If you wanted to make this more consistent, I would do something like this:

```
function getQueriesForTarget(targetId: number) {
  return this.queriesByTarget || [];
}

const queries = this.getQueriesForTarget(targetId);

if (queries.length > 0) {...}

```

But this just adds code, without too much benefit.

Reply:
OK, I'll keep it as-is for now.